plugins/header_rewrite/resources.cc

```diff
@@ -99,6 +104,24 @@ Resources::gather(const ResourceIDs ids, TSHttpHookID hook)
     }
     break;
 
+  case TS_HTTP_TXN_START_HOOK:
+    // Get TCP Info at transaction start
+    if (client_bufp && client_hdr_loc) {
+      TSDebug(PLUGIN_NAME, "\tAdding TXN Start header buffers");
+      bufp    = client_bufp;
+      hdr_loc = client_hdr_loc;
+    }
+    break;
+
+  case TS_HTTP_TXN_CLOSE_HOOK:
+    // Get TCP Info at transaction close
+    if (client_bufp && client_hdr_loc) {
+      TSDebug(PLUGIN_NAME, "\tAdding TXN Close header buffers");
+      bufp    = client_bufp;
```

Comment:
I'd expect headers added on TS_HTTP_TXN_CLOSE_HOOK are available in client *response* header. This code seems like to add headers to client *request* headers.